### PR TITLE
Create load_tissue command

### PR DESCRIPTION
**Pull request by lparsons**
## Summary Change Description

Load tissues command to load tissue records from a `.tsv` file.

## Affected Issue Numbers

- Addresses #350, but has not yet replaced the fixtures

## Code Review Notes

Uses [`transaction.atomic`](https://docs.djangoproject.com/en/3.2/topics/db/transactions/#controlling-transactions-explicitly) to ensure that only files with no errors result in something loaded. Also uses that functionality for a dry-run option.

## Checklist

- [ ] All issue requirements satisfied (or no linked issues)
- [x] [Linting passes](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#linting).
- [x] [Migrations created & committed *(or no model changes)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#migration-process)
- [x] [Tests implemented *(or no code changes)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#test-implementation)
- [x] [All tests pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#quality-control)

### CODE REVIEW

File: DataRepo/management/commands/load_study.py [hunk 1]
```diff
@@ -67,6 +67,13 @@ def handle(self, *args, **options):
             )
             call_command("load_compounds", compounds=compounds_file)
 
+        if "tissues" in study_params:
```
Comment (jcmatese) [inline DataRepo/management/commands/load_study.py:70]: Q: is the study params yaml /config documented in a curator-friendly manner anywhere (wiki)? (or just self-documented in DataRepo/schemas/load_study.yaml or the example `small_obob_study_params.yaml`  ).   Nvm, actually looks like we have a copy [here (loading.yaml)](https://nplcadmindocs.princeton.edu/index.php/TraceBase).  I was just worried about the additional keys, and having them be discoverable by the curator.

File: DataRepo/management/commands/load_tissues.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
+import pandas as pd
+from django.core.management import BaseCommand, CommandError
+
+from DataRepo.utils import DryRun, LoadingError, TissuesLoader
+
+
+class Command(BaseCommand):
+    # Show this when the user types help
+    help = "Loads data from a compound list into the database"
```
Comment (jcmatese) [inline DataRepo/management/commands/load_tissues.py:9]:
```suggestion
    help = "Loads data from a tissue list into the database"
```

File: DataRepo/tests/loading/test_loading_tissues.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        """Test the load_tissue management command with file containing errors"""
+        with self.assertRaisesRegex(
+            CommandError,
+            r"2 errors loading tissue records from .*tissues_with_errors\.tsv - NO RECORDS SAVED",
+        ):
+            call_command(
+                "load_tissues",
+                tissues="DataRepo/example_data/testing_data/tissues/tissues_with_errors.tsv",
+            )
+        # If errors are found, no records shold be loaded
```
Comment (jcmatese) [inline DataRepo/tests/loading/test_loading_tissues.py:39]:
```suggestion
        # If errors are found, no records should be loaded
```

File: DataRepo/utils.py [hunk 4]
```diff
@@ -1800,6 +1800,73 @@ def get_compound_synonym_list_df(cls):
...
+        self.existing = []
+
+    @transaction.atomic
+    def load(self):
+        for index, row in self.tissues.iterrows():
+            try:
+                with transaction.atomic():
+                    name = row["name"]
+                    description = row["description"]
+                    tissue, created = Tissue.objects.get_or_create(name=name)
```
Comment (jcmatese) [inline DataRepo/utils.py:1842]: Q: what is the different between using the `@transaction.atomic` (hadn't seen that before) and the `with transaction.atomic():` context block?  Just wondering if other code that was intended to be "all-or-nothing" is missing something...

**Review by jcmatese — APPROVED** [review]
Looks good.  Could typos and questions...

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -110,7 +110,7 @@ you used when setting up Postgres.
 ### (Optional) Load Some Example Data
 
     python manage.py load_compounds --compounds DataRepo/example_data/consolidated_tracebase_compound_list.tsv
-    python manage.py loaddata DataRepo/fixtures/tissues.yaml
+    python manage.py load_tissues DataRepo/example_data/tissues/tissues.yaml
```
Comment (hepcat72) [inline CONTRIBUTING.md:113]: NB. I still feel like example data is not the best place to store files such as this.  It's not a big deal, but what if we had a "study" (or studies) that contained the example data.  Then loading for dev stuff would be much simpler. And there should be a separate directory for test data.

**Review by hepcat72 — APPROVED** [review]
Nothing blocking.

**Review by fkang-pu — APPROVED** [review]

File: CONTRIBUTING.md (hunk 5 above)
Comment (lparsons) [reply CONTRIBUTING.md:113]: Could not agree more. I've been waiting the the outstanding PRs to get closed before embarking on a full scale moving of the example/testing data. See #172. In fact, here, I've created a "mini-study" with just tissues. The hope would be to create a couple of example studies with a representative set of data and then create some specialized "testing" files with broken things, etc. used just for testing.

File: DataRepo/utils.py (hunk 4 above)
Comment (lparsons) [reply DataRepo/utils.py:1842]: Nothing really different about them, but they are useful in different contexts. The section in the docs on [Controlling Transactions Explicitly](https://docs.djangoproject.com/en/4.0/topics/db/transactions/#controlling-transactions-explicitly) might help explain a bit more. This pattern originated from the example there.

**Pull request merged**